File: code/matlab/lab5.m
clc;
clear all;
close all;
c = 3*(10^8);                                                        % Velocity of light, c
f = 900*(10^6);
D = 1;                                                                   % Maximum Dimension, D
lemda = (c/f);                                                        % Wavelength,, lamda
df = 2*(D^2)/(lemda);                                          % Fraungofer distance, df
pl = -10*log10((lemda^2)/(((4*pi)^2)*(df^2)));  % Path loss, Pl(dB)
fprintf('Fraunhofer distance, Df: %.2f m\n',df)
fprintf('Path Loss PL(dB) : %.2f dB\n',pl);

File: code/python/lab5.py
import math
Operating_frequency = 900 * pow(10,6)
Velocity_of_light = 3*pow(10,8)
Wavelength = round((Velocity_of_light / Operating_frequency), 2)
Max_dimension = 1
Fraunhoffer_distance = round(((2 * pow(Max_dimension, 2)) / Wavelength), 0)
Path_loss = round((-10*math.log10(pow(Wavelength,2)/pow((4*math.pi * Fraunhoffer_distance),2))), 2)

print('Wavelength: ', Wavelength, 'm')
print('Fraunhoffer distance: ', Fraunhoffer_distance, 'm')
print('Path Loss: ', Path_loss,'dB')